app/models/comment.rb

```diff
@@ -1,4 +1,20 @@
 class Comment < ApplicationRecord
-  belongs_to :users
-  belongs_to :posts
+  belongs_to :user, class_name: 'User'
+  belongs_to :post, class_name: 'Post'
+
+  def initialize(*args)
+    super(*args)
+    update_comments_counter if args.any? # Executes the method only if an argument is provided.
+  end
+
+  # Method to update the comments counter for a post.
+  def update_comments_counter
+    post = Post.find_by(id: post_id)
+    counter = if post.comments_counter.nil?
+                1
+              else
+                post.comments_counter + 1
+              end
+    post.update(comments_counter: counter)
+  end
```

Comment:
- [ ] I suggest that you use [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_comments_counter` method every time a new comment has been saved to the database instead of using the initialize method. :relaxed: 
  
  ```ruby
  after_save :update_comments_counter
  ```

- [ ] I suggest that you use the `increment` method for the comments_counter attribute of a post object. This is useful if you want to keep track of the number of comment created on a post without having to recompute the count every time. 
  
  ```ruby
  def update_comments_counter
    post.increment!(:comments_counter)
  end
  ```